Mage/src/main/java/mage/abilities/keyword/PoisonousAbility.java

```diff
@@ -44,4 +44,12 @@ public AddPoisonCountersEffect(int n) {
         setText(n == 1 ? "that player gets a poison counter"
                 : String.format("that player gets %d poison counters", n));
     }
+
+    private AddPoisonCountersEffect(final AddPoisonCountersEffect effect) {
+        super(effect);
+    }
+
+    public AddPoisonCountersEffect copy() {
```

Comment:
Copy must be with `@Override`

Reply:
oops indeed, made that too fast.

Reply:
It's interesting. If it pass your new tests then that tests must be improved to catch this type of error.

Just a theory -- take parent class of the object, if it contains a copy method then call it as a parent object itself.

Reply:
I think I'll go with https://stackoverflow.com/questions/2315445/how-to-quickly-determine-if-a-method-is-overridden-in-java
If there is a copy method, check that it is overriding something.

Reply:
Override annotation is just a flag for the compiler right? It raises an error when isn't a method in the super class with the same signature. Leaving out the annotation is potentially confusing for developers but no impact to code logic.